### PR TITLE
alertmanager: fix accidental cross-service silences

### PR DESCRIPTION
**Pull request by majewsky**

### CODE REVIEW

File: global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml [hunk 1]
```diff
@@ -52,10 +52,16 @@ spec:
         - matchType: "=~"
           name: severity
           value: 'critical|warning'
+        - matchType: "=~"
+          name: context
+          value: '.+'
```
Comment (IvoGoman) [inline global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml:57]:
```suggestion
        - matchType: "!="
          name: context
          value: ''
```

File: global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml [hunk 2]
```diff
@@ -52,10 +52,16 @@ spec:
...
       targetMatch:
         - matchType: "="
           name: severity
           value: 'info'
+        - matchType: "=~"
+          name: context
+          value: '.+'
```
Comment (IvoGoman) [inline global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml:64]:
```suggestion
        - matchType: "!="
          name: context
          value: ''
```

File: global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml [hunk 3]
```diff
@@ -31,17 +31,17 @@ spec:
...
       targetMatch:
         - matchType: "="
           name: severity
           value: 'warning'
-        - matchType: "="
+        - matchType: "=~"
           name: context
           value: '.+'
```
Comment (IvoGoman) [inline global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml:46]:
```suggestion
        - matchType: "!="
          name: context
          value: ''
```

File: global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml [hunk 4]
```diff
@@ -31,17 +31,17 @@ spec:
...
-         value: 'critical'
-       - matchType: "="
-         name: context
-         value: '.+'
+        - matchType: "="
+          name: severity
+          value: 'critical'
+        - matchType: "=~"
+          name: context
+          value: '.+'
```
Comment (IvoGoman) [inline global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml:39]:
```suggestion
        - matchType: "!="
          name: context
          value: ''
```

**Review by IvoGoman — CHANGES_REQUESTED** [review]
I have manually validated the matcher on our alertmanager.
For an Alert **without** the context label it did still match with `context =~ '.*'`. With `context != ''` these alerts are getting filtered out.

File: global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml (hunk 1 above)
Comment (majewsky) [reply global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml]: I'm assuming that means you are confident that `!=` is allowed here. I was going to use that first, but did not find good documentation on which match types are allowed, so I kept to what was already used in the file.

Comment (IvoGoman) [reply global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml]: Yes, you can double check within the link I shared in Company Slack

Comment (viennaa) [reply global/prometheus-alertmanager-operated/templates/am-config-inhibit-rules-base.yaml]: The Matcher is allowed:

https://github.com/prometheus-operator/prometheus-operator/blob/9092aa95a3acfd4c5caedacd0afffd315cf5a154/pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go#L1039-L1044

**Comment by viennaa** [comment]
Can you elaborate, when exactly did these accidental cross service silences happen? I can get the justification out of the name.

**Comment by viennaa** [comment]
One more thing: bump patch version of the chart pls.

**Comment by IvoGoman** [comment]
> Can you elaborate, when exactly did these accidental cross service silences happen? I can get the justification out of the name.

I have cc'ed you on the slack thread

**Comment by majewsky** [comment]
> One more thing: bump patch version of the chart pls.

Done.

**Review by viennaa — APPROVED** [review]
Since context is missing on many alerts, we don't want the unexpected inhibits. Taking suggestion.

**Review by IvoGoman — APPROVED** [review]

**Pull request merged**